applications/GeoMechanicsApplication/custom_processes/apply_constant_boundary_phreatic_surface_pressure_process.hpp

```diff
@@ -161,61 +124,18 @@ class ApplyConstantBoundaryPhreaticSurfacePressureProcess : public Process
     Vector3 mNormalVector;
     double mEqRHS;
 
-///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-
-private:
-
     void calculateEquationParameters()
     {
-        Vector3 v1;
-        Vector3 v2;
-        v1 = mSecondReferenceCoordinate - mFirstReferenceCoordinate;
-        v2 = mThirdReferenceCoordinate - mFirstReferenceCoordinate;
-
-        mNormalVector[0] = v1[1] * v2[2] - v1[2] * v2[1];
-        mNormalVector[1] = v1[2] * v2[0] - v1[0] * v2[2];
-        mNormalVector[2] = v1[0] * v2[1] - v1[1] * v2[0];
-
-        double sum=0;
-        for (unsigned int i=0; i < mNormalVector.size(); ++i)
-        {
-            sum += mNormalVector[i]*mNormalVector[i];
-        }
-        double norm = std::sqrt(sum);
-        if (!(norm > 0.0))
+        const Vector3 v1 = mSecondReferenceCoordinate - mFirstReferenceCoordinate;
+        const Vector3 v2 = mThirdReferenceCoordinate  - mFirstReferenceCoordinate;
+        mNormalVector = MathUtils<double>::CrossProduct(v1, v2);
+        if (norm_2(mNormalVector) <= 0.0)
```

Comment:
The result of `norm_2` can never be negative, or could it? To make the check equivalent to the old code we could change it to:
```suggestion
        if (norm_2(mNormalVector) == 0.0)
```

However, in general it is a bad idea to test real numbers for equality. For now I suggest we accept it, but at some point we may want to introduce a tolerance here (which should be applied consistently across the entire code base). Perhaps a function (e.g. `IsZeroVector`) which checks whether a vector equals the zero vector, taking into account a consistent tolerance value?

Reply:
Indeed as it is a square root it can never be negative. Changed to ==